code/game/turfs/simulated/wall_icon.dm

```diff
@@ -123,9 +130,16 @@
 		I = image(material_icon_base, "[wall_connections[i]]", dir = 1<<(i-1))
 		I.color = base_color
 		add_overlay(I)
-		if(other_connections[i] != "0")
-			I = image(material_icon_base, "other[wall_connections[i]]", dir = 1<<(i-1))
-			I.color = base_color
+		if(paint_color)
+			if("paint0" in icon_states(icon))
```

Comment:
We use a proc to cache icon state checks. `if(check_state_in_icon("paint0, icon))`

Reply:
Now I think about it I'm pretty sure I added unit testing for this so we could remove the state checks entirely due to the overhead they were causing. Will check.

Reply:
See comment on unit test changes below.